### DISCUSSION TITLE
make bin/ipxe.iso EMBED=my.ipxe  , iso can not boot on dell r740

### DISCUSSION THREAD

**Discussion by muziling** [ipxe/ipxe#1008] [General]
I wirte a script my.ipxe, use 
"make bin/ipxe.iso EMBED=my.ipxe"  to build,  generate about 1M size iso,  try to boot on dell r740  and fail.

I download official iso from https://boot.ipxe.org/ipxe.iso,  this  file about 4m size large and can boot on dell r740.

How embed my custom script and build ?

**Comment by NiKiZe** [comment]
When it success in booting, can you go into the iPXE shell using ctrl-b and run ifstat.

Then post here both what your features line says (shown when iPXE starts) and what the first line of ifstat shows?

The size difference is due to your image not being built with EFI binaries.
However your "try to boot and fail" gives no details on how it fails.

**Reply by muziling** [reply]
How to build both embed custom script and  EFI binaries? Ofiicial document not found how to.  
Can not booting ,so  no iPXE shell to debug.

**Reply by NiKiZe** [reply]
You wrote that https://boot.ipxe.org/ipxe.iso can be booted?

Documentation on buildtargets https://ipxe.org/appnote/buildtargets
You will want bin-x86_64-efi in your make command instead of just bin.

To generate a combined iso, see:
https://github.com/ipxe/ipxe/pull/204/commits/79c0173d6df8580ea2e35fc173a5281e65d5321f

**Reply by muziling** [reply]
try to use "make bin/ipxe.iso EMBED=my.ipxe,bin-x86_64-efi/ipxe.efi " , still can not booted.

try to build with
make bin/ipxe.lkrn EMBED=my.ipxe
./util/genfsimg -o combined.iso bin-x86_64-efi/ipxe.efi bin/ipxe.lkrn

this  combined.iso can boot ,but not embed my.ipxe script

**Reply by muziling** [reply]
finally ,use follow two commands ,works.
make bin-x86_64-efi/ipxe.efi EMBED=my.ipxe
./util/genfsimg -o combined.iso bin-x86_64-efi/ipxe.efi bin/ipxe.lkrn

**Reply by muziling** [reply]
Hope generate iso method can be write to document, not just in the commit 79c0173d6df8580ea2e35fc173a5281e65d5321f

**Reply by NiKiZe** [reply]
Are you saying that make bin-x86_64-efi/ipxe.iso EMBED=my.ipxe doesn't work? You only need the genfsimg command if you want to combine legacy pcbios boot with EFI boot.

**Reply by muziling** [reply]
make bin/ipxe.iso EMBED=my.ipxe  doesn't work,  
make bin-x86_64-efi/ipxe.iso EMBED=my.ipxe  works.

**Reply by muziling** [reply]
build with embed, when execute, error output: ping command not found
my.ipxe's line:
ping --count 1 ${netX/gateway} && goto centos7_vnc

ok , NEED TO MODIFY HEADER C FILE

**Reply by NiKiZe** [reply]
Now you are in to different problems, and you get error messages with URLs that describes more or less what you need to do.
